openquake/calculators/event_based.py

```diff
@@ -241,35 +240,51 @@ def save_ruptures(self, ruptures_by_grp_id):
                     self.datastore.extend(
                         ev, numpy.array(events, calc.stored_event_dt))
 
-                # save rup_data
-                if hasattr(ruptures_by_grp_id, 'rup_data'):
-                    trt = self.grp_trt[grp_id]
-                    sm_id = self.sm_by_grp[grp_id]
-                    key = 'rup_data/sm-%04d/%s' % (sm_id, trt)
-                    self.rup_data[trt] = self.datastore.extend(
-                            key, ruptures_by_grp_id.rup_data)
+            # save rup_data
+            if hasattr(ruptures_by_grp_id, 'rup_data'):
+                for grp_id, data in sorted(
+                        ruptures_by_grp_id.rup_data.items()):
+                    key = 'rup_data/grp-%02d' % grp_id
+                    self.rup_data = self.datastore.extend(key, data)
 
     def post_execute(self, result):
         """
         Save the SES collection
         """
+        logging.info('Setting event years')
+        inv_time = int(self.oqparam.investigation_time)
+        numpy.random.seed(self.oqparam.random_seed)
+        for sm in self.datastore['events']:
+            set_random_years(self.datastore['events/' + sm].value, inv_time)
```

Comment:
Good catch. The issue is that I am setting the year only in memory and not in the datastore. I am going to fix that.